### PR TITLE
Use proper signing algorithm for RSA certs

### PR DESCRIPTION
**Pull request by dstiliadis**
Updates on functions for better support of RSA certificates. Most of our code assumes EC everywhere.

### CODE REVIEW

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "f0c27f2e49f53d15ed79ac06d3215403e62aa8f2"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "f2a980737dda13336fa48cd8649da70758cf513d"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "6b27d018f3a0bc8c4600b8092b8a28c0869e00f8"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "a4b4d375f7c9aedb2a52f35bf91d349d8fd2a20a"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "cca8f0442f4db7305c4baffacc6c3ac23626eaff"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "cea08d54feaa5527b1b0e5d0cdbb0e5bbdc83cce"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "a2a24f94d49e83568b8ebc141bbf87f603d171c6"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "e3deddc16b66e254e5ed2b23c808da5778507666"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "a2a24f94d49e83568b8ebc141bbf87f603d171c6"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "e3deddc16b66e254e5ed2b23c808da5778507666"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "1825095f7c2bd1565b800779855e5c39394c4023"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "apocheck",
    "pr-id": "42",
    "commit-sha": "9574c0657cede753fc2058d2c49d02c3e5505b68"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "e3deddc16b66e254e5ed2b23c808da5778507666"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "1825095f7c2bd1565b800779855e5c39394c4023"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "e3deddc16b66e254e5ed2b23c808da5778507666"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "e66c2294c4b52147347646ed527cd9520cf67ae9"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "e3deddc16b66e254e5ed2b23c808da5778507666"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "0ed64d4dff93dd98786586621f53f10b6a789168"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "e3deddc16b66e254e5ed2b23c808da5778507666"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "0ed64d4dff93dd98786586621f53f10b6a789168"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "c6a96f921a20d54e104639c77ff1f2dad725df2c"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "0ed64d4dff93dd98786586621f53f10b6a789168"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "c6a96f921a20d54e104639c77ff1f2dad725df2c"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "a3006d67751b86c288a690a3eb4d56d70640c3c7"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "5a8281ba8c5c08b5f23d33aada389f2eeb7f1ead"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "a3006d67751b86c288a690a3eb4d56d70640c3c7"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "0f41d57d61e88f08ab886d7be01141f932ea6131"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "94f0cb7a640c0a3ef422228050318282a027195d"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "41ef94358a2a2b3a4ad6458d8a9ccce5845db579"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "e11e098eb4b8e89f29db035b058a6eebc853250a"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "0f41d57d61e88f08ab886d7be01141f932ea6131"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "94a512c922646450fc12424f8c911f5e7b8e0e59"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "a51fb2db9391e851822f2daad48a5db6e6232909"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "e11e098eb4b8e89f29db035b058a6eebc853250a"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "3b1ae696c39e777466ff6d8627fc93e15718a3bc"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "d200b9f74ee24be95dd714caf32e06ba37fe599e"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "35df5e6e2801cfa0c05226d1bc5a94788df654a2"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "4e2baae797502489a7f87c2081cbe94ebbf394e2"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "4e2baae797502489a7f87c2081cbe94ebbf394e2"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "d200b9f74ee24be95dd714caf32e06ba37fe599e"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "aa53068a2ef888e9d296a957e7b9249049dca1a9"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "d200b9f74ee24be95dd714caf32e06ba37fe599e"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "a77a9e511a4eb6f8bc5ca350feba8eb26d23cbce"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2b53cfe4d060c6389b688abf7372f16794ff48d1"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "4e2baae797502489a7f87c2081cbe94ebbf394e2"
  }
]
```

File: tglib/csr.go [hunk 1]
```diff
@@ -46,21 +49,33 @@ func LoadCSRs(csrData []byte) ([]*x509.CertificateRequest, error) {
 // GenerateSimpleCSR generate a CSR using the given parameters.
 func GenerateSimpleCSR(orgs []string, units []string, commonName string, emails []string, privateKey crypto.PrivateKey) ([]byte, error) {
 
+	var alg x509.SignatureAlgorithm
+
+	switch reflect.TypeOf(privateKey) {
```
Comment (primalmotion) [inline tglib/csr.go]: why do you need reflect here?

```
 switch privateKey.(type) {
case *ecdsa.PrivateKey:
case *rsa.PrivateKey:
}
```

Comment (dstiliadis) [reply tglib/csr.go]: fixed

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "a44a0357260213d0717816cffeb50e86d05245d2"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2ee08ed1050d6ebfd07806d595fe2e2cc2a1899f"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "40ed76cafab24e0c5c57ea7c9ea9c7d4728c9f6a"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "d200b9f74ee24be95dd714caf32e06ba37fe599e"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "d200b9f74ee24be95dd714caf32e06ba37fe599e"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "a44a0357260213d0717816cffeb50e86d05245d2"
  },
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2ee08ed1050d6ebfd07806d595fe2e2cc2a1899f"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "ee1e16e1471122e56e7488b54827b699612eee1c"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "2ee08ed1050d6ebfd07806d595fe2e2cc2a1899f"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "ee1e16e1471122e56e7488b54827b699612eee1c"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "d64fa124a84af9abff85bcc394b3d92d56b399ac"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "a44a0357260213d0717816cffeb50e86d05245d2"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "oauth",
    "component": "tg",
    "pr-id": "16",
    "commit-sha": "3476a381a1fd71e9942ccf115c7735fa8d648ab7"
  },
  {
    "project": "oauth",
    "component": "backend",
    "pr-id": "448",
    "commit-sha": "ee1e16e1471122e56e7488b54827b699612eee1c"
  },
  {
    "project": "oauth",
    "component": "underwater",
    "pr-id": "67",
    "commit-sha": "d64fa124a84af9abff85bcc394b3d92d56b399ac"
  },
  {
    "project": "oauth",
    "component": "gaia",
    "pr-id": "354",
    "commit-sha": "a44a0357260213d0717816cffeb50e86d05245d2"
  }
]
```

**Pull request merged**